src/test/java/com/agnieszkapawska/flashcards/domain/facades/FlashcardFacadeTest.java

```diff
@@ -0,0 +1,111 @@
+package com.agnieszkapawska.flashcards.domain.facades;
+
+import com.agnieszkapawska.flashcards.FlashcardAndQuestionTagAbstractTests;
+import com.agnieszkapawska.flashcards.domain.dtos.FlashcardSaveDto;
+import com.agnieszkapawska.flashcards.domain.dtos.FlashcardSaveResponseDto;
+import com.agnieszkapawska.flashcards.domain.models.Flashcard;
+import com.agnieszkapawska.flashcards.domain.models.QuestionTag;
+import com.agnieszkapawska.flashcards.domain.services.FlashcardService;
+import com.agnieszkapawska.flashcards.domain.services.QuestionTagService;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import java.util.*;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.*;
+
+public class FlashcardFacadeTest extends FlashcardAndQuestionTagAbstractTests {
+    @Autowired
+    private FlashcardFacade flashcardFacade;
+    @MockBean
+    private FlashcardService flashcardService;
+    @MockBean
+    private QuestionTagService questionTagService;
+    private Flashcard flashcard;
+
+    @Before
+    public void createFlashcard() {
+        flashcard = new Flashcard();
+        flashcard.setId(1L);
+        flashcard.setQuestion("question");
+        flashcard.setAnswer("answer");
+        flashcard.setExplanation("explanation");
+        //questionTag with flashcards
+        QuestionTag questionTag = new QuestionTag();
+        questionTag.setName("home");
+        questionTag.setFlashcards(new HashSet<>(Collections.singletonList(new Flashcard("kubek", "mug"))));
+        flashcard.getQuestionTagsSet().add(questionTag);
+    }
+
+    @Test
+    public void saveFlashcard_ShouldReturnFlashcardSaveResponseDto() {
+        //given
+        FlashcardSaveDto flashcardSaveDto = new FlashcardSaveDto();
+
+        Set<QuestionTag> questionTagSet = new HashSet<>(Arrays.asList(createQuestionTag(1L, "home"),
+                createQuestionTag(2L, "holiday")));
+        flashcardSaveDto.setTagsSet(new HashSet<>(Arrays.asList("home", "holiday")));
```

Comment:
Current flow is:
1. Create flashcardSaveDto
2. Create questionTagSet
3. Configure flashcardSaveDto

Could we swap points 2 and 3? Let's configure object right after creating it.